### PR TITLE
PAY-1440: Pass version info in checkout options

### PR DESCRIPTION
**Pull request by ramth05**
Added version info variables in checkout options.

### CODE REVIEW

File: woo-razorpay.php [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 require_once __DIR__.'/includes/razorpay-webhook.php';
 require_once __DIR__.'/razorpay-sdk/Razorpay.php';
+require_once(ABSPATH.'wp-admin/includes/plugin.php');
```
Comment (harman28) [inline woo-razorpay.php]: Could you remove the parentheses for consistency with the above imports?

File: woo-razorpay.php [hunk 2]
```diff
@@ -568,6 +569,14 @@ private function enqueueCheckoutScripts($data)
             wp_register_script('razorpay_wc_script', plugin_dir_url(__FILE__)  . 'script.js',
                 array('razorpay_checkout'));
 
+            $mod_version = get_plugin_data(plugin_dir_path(__FILE__).'woo-razorpay.php')['Version'];
```
Comment (captn3m0) [inline woo-razorpay.php]: nit: space around periods

File: woo-razorpay.php (hunk 1 above)
Comment (captn3m0) [inline woo-razorpay.php]: nit: no () for require_once.

File: woo-razorpay.php [hunk 3]
```diff
@@ -568,6 +569,14 @@ private function enqueueCheckoutScripts($data)
...
+
+            $data['_'] = array(
+                    'integration'           => 'woocommerce',
+                    'integration_version'   => $mod_version,
```
Comment (harman28) [inline woo-razorpay.php]: Align the rockets 🚀

File: woo-razorpay.php [hunk 4]
```diff
@@ -568,6 +569,14 @@ private function enqueueCheckoutScripts($data)
             wp_register_script('razorpay_wc_script', plugin_dir_url(__FILE__)  . 'script.js',
                 array('razorpay_checkout'));
 
+            $mod_version = get_plugin_data(plugin_dir_path(__FILE__).'woo-razorpay.php')['Version'];
+
+            $data['_'] = array(
```
Comment (harman28) [inline woo-razorpay.php]: I feel it is more appropriate to move this to `getDefaultCheckoutArguments` and not here in `enqueueCheckoutScripts`. Please let me know if I'm missing something.

File: woo-razorpay.php [hunk 5]
```diff
@@ -568,6 +569,14 @@ private function enqueueCheckoutScripts($data)
...
+                    'integration'           => 'woocommerce',
+                    'integration_version'   => $mod_version,
+                    'integration_parent_version' => WOOCOMMERCE_VERSION,
```
Comment (captn3m0) [inline woo-razorpay.php]: Can you confirm if this is defined in all of our supported versions?

File: woo-razorpay.php (hunk 4 above)
Comment (ramth05) [reply woo-razorpay.php]: yes, its fine

File: woo-razorpay.php (hunk 3 above)
Comment (ramth05) [reply woo-razorpay.php]: ok

File: woo-razorpay.php (hunk 1 above)
Comment (ramth05) [reply woo-razorpay.php]: yep

File: woo-razorpay.php (hunk 5 above)
Comment (ramth05) [reply woo-razorpay.php]: This "WOOCOMMERCE_VERSION" const is present in first release as well, so it's supported by all of our plugins versions.

File: woo-razorpay.php (hunk 1 above)
Comment (ramth05) [reply woo-razorpay.php]: ok

File: woo-razorpay.php [hunk 6]
```diff
@@ -491,9 +498,7 @@ protected function createRazorpayOrderId($orderId, $sessionKey)
             }
             catch (Exception $e)
             {
-                $message = $e->getMessage();
-
-                return 'RAZORPAY ERROR: Order creation failed with the message \'' . $message . '\'';
+                return $e;
```
Comment (harman28) [inline woo-razorpay.php]: This will catch the exception and return it. Won't fix the problem of the exception being caught.

Comment (ramth05) [reply woo-razorpay.php]: it's getting caught in parent calling "getRazorpayPaymentParams" function.

Comment (harman28) [reply woo-razorpay.php]: That will also work if you just call `$api->order->create($data)` directly. What is the need for the try-catch and return?

Comment (ramth05) [reply woo-razorpay.php]: It's written to handle the error in proper way and display the message properly.

**Pull request merged**